httputil/request.go

```diff
@@ -172,14 +179,25 @@ func SetAuthType(req *http.Request, res *http.Response) {
 }
 
 func GetAuthType(res *http.Response) string {
-	auth := res.Header.Get("Www-Authenticate")
-	if len(auth) < 1 {
-		auth = res.Header.Get("Lfs-Authenticate")
+
+	authType := GetAuthTypeFromHeader(res, WwwAuthenticateHeader)
+	if authType == BasicAuthType {
+		// Let us check Lfs-Authenticate header to see if server supports NTML auth.
+		authType = GetAuthTypeFromHeader(res, LfsAuthenticateHeader)
 	}
 
-	if strings.HasPrefix(strings.ToLower(auth), "ntlm") {
-		return "ntlm"
+	return authType
+}
+
+func GetAuthTypeFromHeader(res *http.Response, headerName string) string {
+	authHeaders := res.Header[headerName]
+	for i := range authHeaders {
```

Comment:
You can grab the values contained in this slice by using the two-value `range` like so:

```go
for _, auth := range authHeaders {
        if strings.HasPrefix(...) {
        }
}
```